pallets/runtime/tests/src/corporate_actions_test.rs

```diff
@@ -1930,6 +1930,14 @@ fn dist_reclaim_works() {
         assert_ok!(transfer_caa(ticker, owner, other));
         Timestamp::set_timestamp(6);
         let pid = PortfolioId::default_portfolio(other.did);
+        
+        // No custody over portfolio.
+        let custody =
+            |who: User| Custodian::insert(PortfolioId::default_portfolio(other.did), who.did);
+        custody(owner);
+        assert_noop!(reclaim(id, other), PError::UnauthorizedCustodian);
+        custody(other);
```

Comment:
You'll need to test that someone other than `other.did`, who does have custody, can reclaim, and that someone who doesn't have custody, including if `other.did` doesn't over their own portfolio, cannot reclaim. So there are more cases here to consider. The first `NotDistributionCreator` is relevant.

Also, let's check a case where the custodian is not a permissioned external agent (should fail).